nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSSession.java

```diff
@@ -192,7 +192,9 @@
 
     public DBSSession(DBSRepository repository) {
         this.repository = repository;
-        transaction = new DBSTransactionState(repository, this);
+        @SuppressWarnings("resource") // connection closed by DBSTransactionState.close
+        DBSConnection connection = repository.getConnection();
+        transaction = new DBSTransactionState(repository, connection, this);
```

Comment:
I didn't find any other instantiation of DBSTransactionState which makes me think we don't need to change the DBSTransactionState constructor and just do the getConnection into it, wdy?

Reply:
Yes I'll do that